components/kubeconfig-service/pkg/runtime/runtime_kcp.go

```diff
@@ -308,3 +307,27 @@ func cleanConfigMap(coreClientset kubernetes.Interface, userID string, runtimeID
 	}
 	return nil
 }
+
+func GetRawConfig(runtimeID string) ([]byte, error) {
+	coreClientset, err := GetK8sClient()
+	if err != nil {
+		log.Errorf("Failed to get kcp k8s client.")
+		return nil, err
+	}
+	secretList, err := coreClientset.CoreV1().Secrets(KcpNamespace).List(context.Background(), metav1.ListOptions{LabelSelector: fmt.Sprintf("kyma-project.io/runtime-id=%s", runtimeID)})
+	if len((*secretList).Items) == 0 {
+		log.Infof("No secrect found for runtime %s.", runtimeID)
+		return nil, k8serrors.NewNotFound(schema.GroupResource{}, "")
+	} else if err != nil {
+		log.Errorf("Failed to filter secrect for runtime %s: %s", runtimeID, err.Error())
+		return nil, err
+	}
+
+	var secret []byte
+	for _, item := range (*secretList).Items {
```

Comment:
I'm not sure if there will be a situation when label selector used (`kyma-project.io/runtime-id:%s`) will return more than one secret, but I wonder if it at least should be checked and logged as an error or warn. What do you think?

Reply:
A reasonable suggestion, but I don't know if that will happen, since infrastructure-manager should be responsible for preventing such situation. I will log a warn instead of error here.

Reply:
Thanks. IM will not create more than one secret, however if someone manually does that, we don't have a mechanism that will decide which secret is a correct one and currently also just logging an error in such case.